### ISSUE TITLE
Symlinked directories on remote linux boxes - SFTP net drive

### ISSUE THREAD

**Issue by QwertyZW** [atom/atom#12905]
### Prerequisites

Atom version: 1.10.2
OS Version: Windows 10 Pro, Version 1607, Build 14393.222
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode)?
  Yes
- [x] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?
  Yes 
- [x] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?
  Yes, This would be a seperate fuzzy finder issue
- [x] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?
  Yes
- [x] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?
  Probably...
- [x] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?
  Yes
### Description

Sub-issue 1) When using "File -> Open Folder..." to open a folder within a remote filesystem mounted by SFTP Net drive Atom does not see symlinked directories (which File explorer sees) within the directory in question
Sub-issue 2) When opening a symlinked directory within a remote filesystem mounted by SFTP Net drive Atom atom opens a new untitled file.
Sub-issue 3) When using "File -> Open File" to open a file under a symlinked directory within a remote filesystem mounted by SFTP Net drive, Atom reads the file and displays it, but raises "Unable to save file '<Path/to/symlinked/dir>'
EEXIST: file already exists, mkdir '<Path/to/symlinked/dir>'" When attempting to save the file after editting it.
[Description of the bug or feature]

Atom CAN open folders under symlinked directories and files under such folders normally.

In summary, this is a directory structure that you're used to:

```
FileSystem Z:
    -directory1(can be opened normally using "File -> Open Folder..."
    -simlink1(to a directory) (Is not listed under explored) (Is opened as an untitled file using "Open Folder")
        -file1 (can be opened but can't be saved)
        -directory(can be opened normally)
            -file2(can be opened and saved)
```
